clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -526,11 +526,25 @@ scheduler_tmpl_t<RecordType, ReaderType>::init(
                 return res;
         }
     } else {
-        // TODO i#5843: Implement scheduling onto synthetic cores.
-        // For now we only support a single output stream with no deps.
-        if (options_.deps != DEPENDENCY_IGNORE || output_count != 1)
+        // TODO i#5843: Honor deps when scheduling on synthetic cores.
+        if (options_.deps != DEPENDENCY_IGNORE)
             return STATUS_ERROR_NOT_IMPLEMENTED;
-        outputs_[0].cur_input = 0;
+        // TODO i#5843: Implement time-based quanta.
+        if (options_.quantum_unit != QUANTUM_INSTRUCTIONS)
+            return STATUS_ERROR_NOT_IMPLEMENTED;
+        // Assign initial inputs.
+        // TODO i#5843: Once we support core bindings and priorities we'll want
+        // to consider that here.
+        for (int i = 0; i < static_cast<int>(outputs_.size()); i++) {
+            if (i < static_cast<int>(inputs_.size()))
+                set_cur_input(i, i);
+            else
+                set_cur_input(i, -1);
```

Comment:
Use symbolic constant kInvalidInput for -1.

Reply:
Added INVALID_INPUT_ORDINAL and INVALID_OUTPUT_ORDINAL of the new types.